### PR TITLE
Use Github Action

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

### CODE REVIEW

**Comment by koushiro** [comment]
How about using Github Action for CI ?

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+on: [push, pull_request]
+
+name: build
+
+jobs:
+  lint:
+    name: Format and Clippy
+    strategy:
+       matrix:
+         os: [ubuntu-latest, macos-latest]
```
Comment (YangKeao) [inline .github/workflows/rust.yml]: Should we also test it on `windows-latest`? With #15 merged, it (should) can be compiled and tested on windows now.

Comment (koushiro) [reply .github/workflows/rust.yml]: If we want to test it on `windows-latest`, we also need to implement `register_signal_handler` for windows. 
Currently, we use `nix` for handling signal, it obviously does not support windows platform.

Comment (YangKeao) [reply .github/workflows/rust.yml]: Oh! I forgot. The `setitimer` syscall also couldn't be used on windows :(

**Pull request merged**